src/include/catalog/catalog_accessor.h

```diff
@@ -320,7 +320,7 @@ class EXPORT CatalogAccessor {
   proc_oid_t CreateProcedure(const std::string &procname, language_oid_t language_oid, namespace_oid_t procns,
                              const std::vector<std::string> &args, const std::vector<type_oid_t> &arg_types,
                              const std::vector<type_oid_t> &all_arg_types,
-                             const std::vector<postgres::ProArgModes> &arg_modes, type_oid_t rettype,
+                             const std::vector<postgres::PgProc::ProArgModes> &arg_modes, type_oid_t rettype,
```

Comment:
I'm currently leaning towards `PgProc::ArgModes`. I'm inclined to keep the Pg in PgProc because it is meant to be the pg_proc table from postgres -- proc seems ambiguous and might refer to the actual Proc itself (that you mentioned we might want to have as its own thing, elsewhere)

Reply:
Pushed but leaving unresolved.

Reply:
I'm fine with your argument on leaving it as `PgProc`, so go with either way and consider this resolved.